kernel/src/process.rs

```diff
@@ -1096,35 +1106,54 @@ impl<C: Chip> ProcessType for Process<'a, C> {
             // A null buffer means pass in `None` to the capsule
             Ok(None)
         } else if self.in_app_owned_memory(buf_start_addr, size) {
+            // Capture that we've verified this pointer
+            let buf_start = unsafe { NonNull::new_unchecked(buf_start_addr as *mut u8) };
```

Comment:
Nit: I'd say that this `unsafe` is not needed either, because a previous if is checking that `buf_start_addr` is not null. Which could be rewritten as (some variant of):

```rust
match NonNull::new(buf_start_addr) {
  None => { ... }
  Some(buf_start) => {
    if self.in_app_owned_memory(buf_start, size) {
      ...
    }
  }
}
```

Reply:
Done